src/coreclr/tools/aot/WhyDgml/WhyDgml.cs

```diff
@@ -56,10 +60,20 @@ static void Main(string[] args)
         }
 
         string goal = args[1];
-        if (!nameToNode.ContainsKey(goal))
-            Console.WriteLine($"No such node: '{goal}'.");
-        else
-            Dump(nameToNode[goal]);
+        int matchedCount = 0;
+        foreach (var key in nameToNode.Keys)
+        {
+            var isMatched = Regex.IsMatch(key, WildCardToRegular(goal));
+            if (isMatched)
+            {
+                matchedCount++;
+                Console.WriteLine($"**** Match {matchedCount} ****");
+                Dump(nameToNode[key]);
+            }
+        }
```

Comment:
Could this just go to the old `if (!Contains)` branch so that the case when this isn't a regex is still fast?

There's no point in enumerating the keys with a regex when an exact match was already supplied. Some node names (types) have actual `?` in them that will make them look like a regex but they're not regex.

```csharp
        if (!nameToNode.ContainsKey(goal))
            // put new code here
        else
            Dump(nameToNode[goal]);
```